### PR TITLE
Column sorting usability 180887524

### PR DESCRIPTION
**Pull request by MLumme**
Task here https://www.pivotaltracker.com/story/show/180887524, wrote logic to put sorting changes reversed into the sorting array.

### CODE REVIEW

**Review by olzraiti — CHANGES_REQUESTED** [review]
At least the following bug:

1. goto https://180887524.dev.laji.fi/observation/list
2. click taxon column
-> doesn't sort

~I'm assuming that's something about the component having `OnPush` as the change detection method (which is good), and the `updateSorts` returns `this.sorts`, which is immutably edited, but it's still the same array instance and thus the change detection fails even when it's manually called to detect changes. Making the return points of `updateSorts()` to always make a copy of `this.sorts` before returning could fix this?~

EDIT: the above was wrong - `updateSorts` returns always a new instance so the hypothesis isn't right.

File: projects/laji/src/app/shared-modules/datatable/datatable/datatable.component.ts [hunk 1]
```diff
@@ -287,12 +287,43 @@ export class DatatableComponent implements AfterViewInit, OnInit, OnChanges, OnD
   }
 
   onSort(event) {
-    this.sorts = event.sorts;
-    this.sortRows(event.sorts);
+    this.sorts = this.updateSorts(event);
+    this.sortRows(this.sorts);
     this.changeDetectorRef.detectChanges();
+    event.sorts = this.sorts;
```
Comment (olzraiti) [inline projects/laji/src/app/shared-modules/datatable/datatable/datatable.component.ts:293]: Is this necessary? Altering function parameters in mutable fashion is shady business.

Comment (MLumme) [reply projects/laji/src/app/shared-modules/datatable/datatable/datatable.component.ts:293]: Actually not, I should've put a new object into event emitter that is the new sort and rest from destructuring the event, I'll fix it. And I'll try to figure out what's going on with that bug.

Comment (MLumme) [reply projects/laji/src/app/shared-modules/datatable/datatable/datatable.component.ts]: It's fixed now, I had set wrong column property as sort prop, which worked most of the time by accident but not for taxon.

**Review by olzraiti — APPROVED** [review]
Thanks! :+1:

**Pull request merged**